site/content/extend/plugins/webapp/best-practices.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Best Practices
+date: 2019-06-02T00:00:00-00:00
+subsection: Web App Plugins
+weight: 0
+---
+
+This page provides best practices for web app plugins.
+
+## User Experience
+
+### Plugin placement in the user interface
+
+1. Actions that apply to specific channels
+- Recommendation: Have your plugin register the actions to [the channel header](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerChannelHeaderButtonAction). This makes it quickly accessible by users and the actions apply to the channel they're viewing. You can additionally [register a slash command](https://developers.mattermost.com/extend/plugins/server/reference/#API.RegisterCommand) on the server-side to take channel-specific actions.
+- Examples: Jira project subscriptions to a channel; Zoom meeting posts to a channel
+
+2. Actions that apply to specific messages
+- Recommendation: Have your plugin register a [post dropdown menu component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostDropdownMenuComponent) with some text, an icon and an action function. This adds your action to the "More Actions" post menu dropdown and gives easy access to your app.
+- Examples: Create or attach to Jira issue from a message; Copy a message to another channel; report an inappropriate message
+
+3. Actions related to files or images
+- Recommendation: Have your plugin register a [file upload method](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerFileUploadMethod) with some text, icon and an action function. This adds your action to the file upload menu.
+- Examples: File sharing from OneDrive or GDrive; Draw plugin for sketches
+
+4. Actions that apply to specific users
+- Recommendation: Have your plugin register a [popover user actions component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPopoverUserActionsComponent). This adds your action to the user profile popover. 
+- Examples: ..
+
+5. Attributes added for a user
```

Comment:
Feels like we could roll this up into the previous section, and just use the single example but reference both API methods.